reference/monitoring-metrics: update the TiDB metrics (#1435)
pingcap/docs: dev/reference/key-monitoring-metrics/tidb-dashboard.md

If you use Ansible to deploy the TiDB cluster, the monitoring system is deployed at the same time. For more information, see [TiDB Monitoring Framework Overview](/how-to/monitor/overview.md).

> The Grafana dashboard is divided into a series of sub dashboards which include Overview, PD, TiDB, TiKV, Node\_Exporter, Disk Performance, and so on. The TiDB dashboard consists of the TiDB panel and the TiDB Summary panel, and what displays on the TiDB Summary panel also displays on the TiDB panel. A lot of metrics are there to help you diagnose.

Author: > I remember that `Node_exporter` is exactly what's displayed on the dashboard.

When @liubo0127 was reviewing the [TiDB Cluster Alerting Metrics Guide - Google 文档](https://docs.google.com/document/d/1yiv4yRWQ-skuHp1RcagDCMt8bFqSmkDb4I1gdBsS_JU/edit#), he suggested I use "Node_Exporter" instead of "Node_exporter". Please help confirm this name usage. @lonng If `Node_exporter` is what's displayed on the dashboard, I think we should consistently use it in our documents.

> The Grafana dashboard is divided into a series of sub dashboards which include Overview, PD, TiDB, TiKV, Node\_Exporter, Disk Performance, and so on. The TiDB dashboard consists of the TiDB panel and the TiDB Summary panel, and what displays on the TiDB Summary panel also displays on the TiDB panel. A lot of metrics are there to help you diagnose.

Author: Should I use "Node_Exporter" or "Node_exporter"? PTAL @lonng

> The Grafana dashboard is divided into a series of sub dashboards which include Overview, PD, TiDB, TiKV, Node\_Exporter, Disk Performance, and so on. The TiDB dashboard consists of the TiDB panel and the TiDB Summary panel, and what displays on the TiDB Summary panel also displays on the TiDB panel. A lot of metrics are there to help you diagnose.

Reviewer: The name of it in Grafana dashboard is `Node_exporter`, it's better to keep them consistent.

> The Grafana dashboard is divided into a series of sub dashboards which include Overview, PD, TiDB, TiKV, Node\_Exporter, Disk Performance, and so on. The TiDB dashboard consists of the TiDB panel and the TiDB Summary panel, and what displays on the TiDB Summary panel also displays on the TiDB panel. A lot of metrics are there to help you diagnose.

Author: Got it, thanks~

    - Failed Query OPM Detail: the statistics of error types (such as syntax errors and primary key conflicts) according to the errors happening when executing SQL statements in the entire cluster
    - Internal SQL OPS: the statistics of the executed SQL statements within TiDB

- Server
    - Uptime: the runtime of TiDB
    - Memory Usage: the statistics of memory usage of different TiDB instances
    - CPU Usage: the statistics of CPU usage of different TiDB instances
    - Connection Count: the number of clients connected to each TiDB instance
    - Open FD Count: the statistics of opened file descriptors of different TiDB instances
    - Goroutine Count: the number of Goroutines of different TiDB instances
    - Go GC Duration: the statistics of GC time of different TiDB instances
    - Go Threads: the number of threads of different TiDB instances
    - Go GC Count: the number of times that GC is executed on different TiDB instances
    - Go GC CPU Usage: the statistics of GC CPU usage of different TiDB instances
    - Events OPM: the statistics of key events, such as "start", "close", "graceful-shutdown","kill", "hang", and so on

>     - Keep Alive OPM: the number of times that the monitor refreshes every minute on different TiDB instances

Reviewer: ```suggestion
    - Keep Alive OPM: the number of times that the metrics refresh every minute on different TiDB instances
```
I'm not sure which one is better.

Author: Can I use "the monitoring metrics refresh" here?

Author: I think we can use "the monitor display is refreshed" or "the metrics are refreshed" here. (see [Refresh metrics](http://dcx.sap.com/1200/en/relayserver/monitor-refresh-metrics.html))

Reviewer: Prefer to `the metrics are refreshed`.

    - Prepare Statement Count: the number of `Prepare` statements that are executed on each TiDB instance and the total count of them

>     - Time Jump Back OPS: the number of times that the time jumps back every second on different TiDB instances

Reviewer: `jumps back => rewind`? (not sure which one is better)

Author: "rewind" is better. Thanks for pointing it out.

    - Pseudo Estimation OPS: the number of the SQL statements optimized using pseudo statistics
    - Dump Feedback OPS: the number of stored statistical Feedbacks
    - Update Stats OPS: the statistics of using Feedback to update the statistics information

>     - Significant Feedback: the number of significant Feedback that update the statistics information

Reviewer 2: ```suggestion
    - Significant Feedback: the number of significant Feedback pieces that update the statistics information
```

- Meta
    - AutoID QPS: AutoID related statistics, including three operations (global ID allocation, a single table AutoID allocation, a single table AutoID Rebase)
    - AutoID Duration: the time consumed by AutoID related operations
    - Meta Operations Duration 99: the latency of Meta operations

- GC
    - Worker Action OPM: the statistics of GC related operations, including `run_job`, `resolve_lock`, and `delete\_range`
    - Duration 99: the time consumed by GC related operations
    - GC Failure OPM: the number of failed GC related operations
    - Action Result OPM: the number of results of GC-related operations
    - Too Many Locks Error OPM: the number of the error that GC clears up too many locks

- Batch Client

>     - Pending Request Count by TiKV: the number of Batch messages that are pending for processing

Reviewer 2: It seems that the phrase `pending for` is incorrect or is seldom used. Ref: [Dictionary explanation](https://www.merriam-webster.com/dictionary/pending) and [discussion about this word](https://ell.stackexchange.com/questions/218701/is-pending-for-her-return-grammatical).

```suggestion
    - Pending Request Count by TiKV: the number of Batch messages that are pending processing
```

>     - Wait Duration 95: the waiting time of Batch messages that are pending for processing

Reviewer 2: ```suggestion
    - Wait Duration 95: the waiting time of Batch messages that are pending processing
```
